### PR TITLE
docs: Add further user documentation for SNS message filtering

### PR DESCRIPTION
**Pull request by MitchellPaff**

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -249,15 +249,19 @@ $ aws lambda invoke --function-name import-status \
...
 specific dataset or specific statuses. Included in the example is all the valid statuses.
 
+The Geostore will store a dataset in a top level directory name with format of (dataset
+title)-(dataset id) You can filter SNS topics for a dataset by providing the entire dataset
+directory name or by providing just the title in a 'prefix' object. Examples below.
+
 ```json
 {
-  "dataset_id": ["01F9ZGZ4EGEZ9R4Z48VNQHKB0Z", "01F9ZFRK12V0WFXJ94S0DHCP65"],
+  "dataset_id": [{ "prefix": "Taranaki" }, "Taranaki_2020-01F9ZFRK12V0WFXJ94S0DHCP65"],
```
Comment (l0b0) [inline USAGE.md:264]: What's going on here? Are we actually returning a `List[Union[Dict[str, str], str]]`?

File: USAGE.md [hunk 2]
```diff
@@ -249,15 +249,19 @@ $ aws lambda invoke --function-name import-status \
 
 ## Receive Import Status updates by subscribing to our AWS SNS Topic
 
-The ARN of our SNS Topic is `arn:aws:sns:ap-southeast-2:715898075157:geostore-import-status` which
-you may choose to subscribe to.
+The ARN of our SNS Topic is `arn:aws:sns:ap-southeast-2:<YOUR_ACCOUNT_ID>:geostore-import-status`
```
Comment (billgeo) [inline USAGE.md:252]: Isn't this supposed to say <GEOSTORE_AWS_ACCOUNT_ID>?

File: USAGE.md [hunk 3]
```diff
@@ -300,7 +304,7 @@ See
   "id": "3afc69d1-1291-6372-ae76-9a9c11ee35c4",
   "detail-type": "Step Functions Execution Status Change",
   "source": "aws.states",
-  "account": "715898075157",
+  "account": "<YOUR_ACCOUNT_ID>",
```
Comment (billgeo) [inline USAGE.md:307]: Same here

File: USAGE.md [hunk 4]
```diff
@@ -269,7 +273,7 @@ contain a JSON string with specific details regarding the Step Function Executio
 {
   "Type": "Notification",
   "MessageId": "xxxx-xxxx-xxxx",
-  "TopicArn": "arn:aws:sns:ap-southeast-2:715898075157:geostore-import-status",
+  "TopicArn": "arn:aws:sns:ap-southeast-2:<YOUR_ACCOUNT_ID>:geostore-import-status",
```
Comment (billgeo) [inline USAGE.md:276]: Same here.

File: USAGE.md (hunk 1 above)
Comment (billgeo) [reply USAGE.md:264]: Also a little confused. I would have thought it would be something like
`"dataset_id": [{ "prefix": "Taranaki" }]`

**Review by billgeo — CHANGES_REQUESTED** [review]

File: USAGE.md (hunk 1 above)
Comment (MitchellPaff) [reply USAGE.md:264]: We're  not returning anything. This is an example json policy that can be used to filter the SNS messages - it has one example where a prefix is used one where the whole string is matched.

**Review by billgeo — APPROVED** [review]
LGTM. Thanks.

**Review by billgeo — CHANGES_REQUESTED** [review]
Actually the account ID also appears further down in the  example.
Can you make sure there sensitive info anywhere.

**Review by billgeo — APPROVED** [review]
Sorry, ignore that last comment. You changed it.

**Pull request merged**